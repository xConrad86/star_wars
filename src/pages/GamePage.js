import React, { useState, useEffect } from "react";
import { isEmptyObject } from "../utils/Functions";
import { TrinityRingsSpinner } from "react-epic-spinners";
import GameView from './../components/Game/GameView'
import "./styles.scss";

const GamePage = () => {
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState({});

  const handleChangeLoading = (isLoading) => {
    setIsLoading(isLoading)
  }

  const handleError = (error) => {
    setError(error)
  }

  useEffect(() => {  
    let timer;     

    timer = setTimeout(() => {
        handleChangeLoading(false)
    }, 3000);            

    return () => clearTimeout(timer);        
  }, [isLoading])            


  const getAppState = () => {
    if (isLoading) {
      return (
        <React.Fragment>
          <TrinityRingsSpinner          
            size={250}
            className="game-page-spinner"            
          />
          <h1 id="title-wait">
            Please wait... Drawing card for the players            
          </h1>
        </React.Fragment>
      );
    } else {
      return <div>{error.error}</div>;
    }
  };

  return (
    <React.Fragment>
      {isLoading || isEmptyObject(error) === false ? (
        getAppState()
      ) : (
        <GameView handleChangeLoading={handleChangeLoading} handleError={handleError} 
        />
      )}
    </React.Fragment>
  );
};

export default GamePage;
