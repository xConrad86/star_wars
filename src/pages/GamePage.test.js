import React from 'react';
import GamePage from './GamePage.js';
import {shallow, mount} from 'enzyme'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom';



describe('Gamepage component test', () => {
    let wrapper;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init) => [init, setState]); 

    beforeEach(() => {
        wrapper = mount(<GamePage />);
      });
    
    afterEach(() => {
        jest.clearAllMocks();
    });

    it("Display and compare the html of wrapper", () => {     
        expect(wrapper.html()).toMatchSnapshot();
        wrapper.find('title-wait').exists()
     });
     
     it('Render the page after loading the data ', async () => {                                      
            const promise = () => {
              return new Promise(resolve => {
                jest.setTimeout(() => {
                    act(() => {
                        wrapper.mount()
                        wrapper.update();
                      });
                  
                  resolve(wrapper);
                }, 10000);
              });
            };
            return promise().then((res) => {
              console.log(res.debug());
              expect(wrapper.html()).toMatchSnapshot();
            });                                
    }, 30000);
   
    
  
})