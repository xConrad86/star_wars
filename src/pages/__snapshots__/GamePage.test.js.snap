// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gamepage component test Display and compare the html of wrapper 1`] = `"<div size=\\"250\\" color=\\"#fff\\" class=\\"sc-fFeiMQ IEkuh trinity-rings-spinner\\" style=\\"position: absolute; left: 45%; top: 40%; color: white;\\"><div class=\\"circle circle1\\"></div><div class=\\"circle circle2\\"></div><div class=\\"circle circle3\\"></div></div><h1 id=\\"title-wait\\" style=\\"text-align: center;\\">Please wait... Drawing card for the players</h1>"`;
